webservice/static/js/main.js

```diff
@@ -177,6 +177,34 @@ function showResult(imgName) {
     });
 }
 
+function waitAndShowResult(imgName) {
+    console.log('waitAndShowResult', imgName)
+    $.ajax({
+        type: "POST",
+        url: "/result/",
+        data: JSON.stringify({
+            'name': imgName
+        }),
+        success: function (response) {
+            console.log('waitandshowresultsuccess');
+            let status = response['status'];
+            if (status == 'finished') {
+                showResult(imgName);
+            }
+            else{
+                setTimeout(waitAndShowResult(imgName), 1000);
```

Comment:
this doesn't work the way I assume you intended it, i.e. it doesn't wait 1000ms but executes the function immediately.

you probably want something like this instead:

-  setTimeout(waitAndShowResult(imgName), 1000);
+ function helper () {
+     waitAndShowResult(imgName);
+  }
+  setTimeout(helper, 1000);

Reply:
makes sense to me, would have another try